corehq/motech/fhir/tasks.py

```diff
@@ -394,5 +398,252 @@ def create_parent_indices(
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        json_response = response.json()
+        entries = json_response['entry']
+        for entry in entries:
+            appointments.append(entry)
+    elif response.status_code >= 400:
+        response.raise_for_status()
+    return appointments
+
+
+def convert_date_and_time_to_utc_timestamp(date, time):
+    date_time = date + "T" + time
+    utc_zone = dateutil.tz.gettz('UTC')
+    # Hardcoded for MGH study
+    local_zone = dateutil.tz.gettz('America/New_York')
+    local_datetime = datetime.fromisoformat(date_time)
+    local_datetime = local_datetime.replace(tzinfo=local_zone)
+    utc_datetime = local_datetime.astimezone(utc_zone)
+    utc_iso_format = utc_datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+    return utc_iso_format
+
+
+def convert_utc_timestamp_to_date_and_time(utc_timestamp):
+    utc_zone = dateutil.tz.gettz('UTC')
+    local_zone = dateutil.tz.gettz('America/New_York')
+    utc_datetime = datetime.fromisoformat(utc_timestamp.replace('Z', ''))
+    utc_datetime = utc_datetime.replace(tzinfo=utc_zone)
+    local_datetime = utc_datetime.astimezone(local_zone)
+    date = local_datetime.strftime('%Y-%m-%d')
+    time = local_datetime.strftime('%H:%M')
+
+    return date, time
+
+
+def sync_all_appointments_domain(domain):
+    try:
+        access_token = request_epic_access_token()
+    except Exception:
+        return None
+    # get all patient case ids for domain
+    patient_case_ids = CommCareCase.objects.get_open_case_ids_in_domain_by_type(domain, 'patient')
+    # get all patient cases
+    patient_cases = CommCareCase.objects.get_cases(patient_case_ids)
+
+    # get extension (appointment) cases ids, for all patients
+    appointment_case_ids = CommCareCaseIndex.objects.get_extension_case_ids(
+        domain, patient_case_ids, False, 'appointment')
+    # get appointment cases in commcare
+    appointment_cases = CommCareCase.objects.get_cases(appointment_case_ids)
+
+    # get fhir ids for appointments currently in commcare
+    appointment_fhir_ids = [
+        appointment_case.get_case_property('fhir_id') for appointment_case in appointment_cases
+    ]
+
+    for patient in patient_cases:
+        patient_helper = CaseHelper(case=patient, domain=domain)
+        patient_fhir_id = patient.get_case_property('patient_fhir_id')
+        if not patient_fhir_id:
+            given = patient.get_case_property('given_name')
+            family = patient.get_case_property('family_name')
+            birthdate = patient.get_case_property('birthdate')
+            patient_fhir_id = get_patient_fhir_id(given, family, birthdate, access_token)
+
+            if patient_fhir_id is not None:
+                patient_helper.update({'properties': {
+                    'patient_fhir_id': patient_fhir_id,
+                }})
```

Comment:
If `patient_fhir_id is None` you can skip the rest of the loop
```suggestion
                }})
            else:
                continue
```

Reply:
changed here https://github.com/dimagi/commcare-hq/pull/35125/commits/1d51e4730d26fc92e9aac5542f35fe7fd7f925e6